consultation_analyser/consultations/models.py

```diff
@@ -80,7 +81,72 @@ class Meta:
         ]
 
 
+@dataclass
+class MultipleChoiceQuestionStats:
+    question: str
+    counts: dict[str, int]
+    percentages: dict[str, int]
+    has_multiple_selections: bool
+
+
 class Question(UUIDPrimaryKeyModel, TimeStampedModel):
+    MULTIPLE_CHOICE_COUNTS_QUERY = """
+        SELECT question_id as id, elem ->> 'question_text' as question, option, COUNT(*) as count
+        FROM    consultations_answer,
+                jsonb_array_elements(multiple_choice) AS elem,
+                jsonb_array_elements_text(elem -> 'options') AS option
+        WHERE question_id = %s
+        GROUP BY question_id, option, question
+        ORDER BY question, option ASC
+    """
+
+    # until we explicitly annotate these questions, use the presence
+    # of multiple selections to infer whether they're supported
+    HAS_MULTIPLE_SELECTIONS_QUERY = """
+        SELECT EXISTS (
+            SELECT 1
+            FROM consultations_answer,
+                 jsonb_array_elements(multiple_choice) AS elem
+            WHERE question_id = %s
+            AND elem ->> 'question_text' = %s
+            AND jsonb_array_length(elem -> 'options') > 1
+        ) AS has_multiple_selections;
+    """
+
+    def multiple_choice_stats(self) -> list[MultipleChoiceQuestionStats]:
+        counts = type(self).objects.raw(self.MULTIPLE_CHOICE_COUNTS_QUERY, params=[str(self.id)])
+
+        values = [
+            {"question": count.question, "option": count.option, "count": count.count}
+            for count in counts
+        ]
+
+        output = []
+        for q, stats in itertools.groupby(values, lambda group: group["question"]):
+            statsl = list(stats)  # necessary as we need to iterate over this generator twice
+
+            counts = {opt["option"]: opt["count"] for opt in statsl}
+            total = sum(counts.values())
```

Comment:
Is this the right total? What should the denominator be? First instinct would be that the denominator should be the total number of responses.
Though perhaps it only makes sense to have counts.

E.g. If the question was "Which colours do you like?" and the options were "red", "blue", "green". And there were 3 responses: ["blue", "red"], ["blue"], ["green", "red"]. 
I think the table would be:
Blue 2 67%
Red 2 67%
Green 1 33 %

Reply:
Discussed - percentages to be removed.

Reply:
(Removed in this case only.)